src/builtins/sets.rs

```diff
@@ -56,7 +56,7 @@ fn intersection_of_set_of_sets(
         };
 
         if first {
-            res = (**s).clone();
+            res.clone_from(&(**s));
```

Comment:
I think this can be just `res.clone_from(s)` it should auto-dereference.

Reply:
Great suggestion!